subsys/instrumentation/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+# Copyright 2023 Linaro
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_sources_ifdef(CONFIG_INSTRUMENTATION
+  common/instr_common.c
+  timestamp/timestamp.c
+  transport/uart.c
+)
+
+zephyr_sources_ifdef(CONFIG_INSTRUMENTATION_MODE_CALLGRAPH ringbuffer/ringbuffer.c)
+
+if(CONFIG_INSTRUMENTATION)
+  if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+    zephyr_sources(handlers/instr_handlers_gcc.c)
+  else()
+    message(FATAL_ERROR "CONFIG_INSTRUMENTATION unsupported by this compiler.")
```

Comment:
please don't use `FATAL_ERROR` for user config mistakes.
FATAL_ERROR prevents users from (re-)running menuconfig and fix their mistake.

See also: https://github.com/zephyrproject-rtos/zephyr/issues/9573

Reply:
Changed to warning